.ci/auto-gen.sh

```diff
@@ -0,0 +1,115 @@
+#!/bin/bash
+
+set -x
+. "$(dirname $0)/variables.sh"
+
+
+autogen_clear() {
+	DIR="$1"
+	
+	FILES=${DIR}/*
+	for FILE in $(ls $FILES); do
+		if [ -d $FILE ]; then
+			autogen_subdir_clear $FILE
+		fi
+	done				
+}
+
+autogen_subdir_clear() {
+    DIR="$1"
+
+    rm -f ${DIR}/*.go
+}
+
+mocks_clear() {
+    for DIR in $SRC;
+    do
+        MOCKS=${DIR}/*_mock.go
+        if ls $MOCKS &> /dev/null; then
+            for FILE in $(ls $MOCKS);
+            do
+                rm $FILE
+            done
+        fi
+    done
+}
+
+autogen_subdir_cleanup() {
+	DIR="$1"
+
+	FILE=${DIR}/*
+	for FILE in $(ls FILES); do
+		if [ -d $FILE ]; then
+			autogen_subdir_cleanup $FILE
+		else
+			file_cleanup $FILE $DIR
+		fi
+	done
+}
+
+autogen_cleanup() {
+    DIR="$1"
+
+    FILES=${DIR}/*
+    for FILE in $(ls $FILES);
+    do
+				if [ -d $FILE ]; then
+					autogen_subdir_cleanup $FILE
+				fi
+    done
+}
+
+mocks_cleanup() {
+    for DIR in $SRC;
+    do
+        MOCKS=${DIR}/*_mock.go
+        if ls $MOCKS &> /dev/null; then
+            for FILE in $(ls $MOCKS);
+            do
+                file_cleanup $FILE $DIR
+            done
+        fi
+    done
+}
+
+file_cleanup() {
+    FILE="$1"
+    DIR="$2"
+
+    # NB(xichen): there is an open issue (https://github.com/golang/mock/issues/30)
+    # with mockgen that causes the generated mock files to have vendored packages
+    # in the import list. For now we are working around it by removing the vendored
+    # path. Also sed -i'' does not work with BSD sed shipped with OS X, whereas
+    # sed -i '' doesn't work with GNU sed, so we work around it by redirecting to a
+    # temp file first and moving it back later.
+    sed "s|$VENDOR_PATH/||" $FILE > $FILE.tmp && mv $FILE.tmp $FILE
+    
```

Comment:
Yeah I had to make updates to these to support multiple proto packages and
having them spread everywhere was a total pain
On Fri, Aug 26, 2016 at 4:29 PM xichen2020 <notifications@github.com> wrote:

> In .ci/auto-gen.sh
> <https://github.com/m3db/m3storage/pull/5#discussion_r76483186>:
>
> > +        fi
> > +    done
> > +}
> > +
> > +file_cleanup() {
> > +    FILE="$1"
> > +    DIR="$2"
> > +
> > +    # NB(xichen): there is an open issue (https://github.com/golang/mock/issues/30)
> > +    # with mockgen that causes the generated mock files to have vendored packages
> > +    # in the import list. For now we are working around it by removing the vendored
> > +    # path. Also sed -i'' does not work with BSD sed shipped with OS X, whereas
> > +    # sed -i '' doesn't work with GNU sed, so we work around it by redirecting to a
> > +    # temp file first and moving it back later.
> > +    sed "s|$VENDOR_PATH/||" $FILE > $FILE.tmp && mv $FILE.tmp $FILE
> > +
>
> We'll also need to strip the $GOPATH in source files so the source file
> path doesn't change when a different person runs make mock-gen:
> https://github.com/m3db/m3cluster/blob/master/.ci/auto-gen.sh#L79
>
> Not for this diff but perhaps it's worth moving these bash scripts to a
> separate build module after all and have them vendored in
> m3db/m3cluster/m3storage so we don't have to copy it everywhere.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/m3db/m3storage/pull/5/files/54ef78fe53421c3c1198ade1cf7eac5c8d573520#r76483186>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAX-WbrErl0rVybIi5kRzlpg8avFWIcuks5qj0y_gaJpZM4Jubit>
> .
>